### PR TITLE
Fix dev_appserver arguments syntax

### LINKED ISSUE #121: The JVM flags are incorrectly set for the dev app server run command
When adding the arguments to the dev app server run command, the jvm flags are set as "--jvm_flag -Dappengine.user.timezone=UTC" instead of "--jvm_flag=-Dappengine.user.timezone=UTC", which causes the following error:

usage: dev_appserver.py [-h] [-A APP_ID] [--host HOST] [--port PORT]
                        [--admin_host ADMIN_HOST] [--admin_port ADMIN_PORT]
                        ...
                        yaml_or_war_path [yaml_or_war_path ...]
dev_appserver.py: error: argument --jvm_flag: expected one argument

### PR DESCRIPTION
**Pull request by meltsufin**
Fixes #121.

@nbashirbello @loosebazooka PTAL

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/args/Args.java [hunk 1]
```diff
@@ -52,14 +52,13 @@
   }
 
   /**
-   * @return [--name, value1, --name, value2, ...] or [] if value=null.
+   * @return [--name=value1, --name=value2, ...] or [] if value=null.
    */
-  static List<String> strings(String name, List<String> values) {
+  static List<String> stringsEq(String name, List<String> values) {
```
Comment (elharo) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/args/Args.java]: please use  amore complete method name. I'd guess this is testing strings for equality but it doesn't seem to be.

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/args/Args.java [hunk 2]
```diff
@@ -76,6 +75,16 @@
   }
 
   /**
+   * @return [--name=value] or [] if value=null.
+   */
+  static List<String> integerEq(String name, Integer value) {
```
Comment (elharo) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/args/Args.java]: ditto

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/args/DevAppServerArgs.java [hunk 3]
```diff
@@ -16,32 +16,49 @@
...
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 /**
  * Command Line argument helper for dev_appserver based command.
  */
 public class DevAppServerArgs {
 
+  public static final Set<String> NON_EQ_PARAM_NAMES = new HashSet<>(Arrays.asList(
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/args/DevAppServerArgs.java]: this is kinda weird, does = not work everywhere?

Comment (elharo) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/args/DevAppServerArgs.java]: danger: mutable public field

Comment (elharo) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/args/DevAppServerArgs.java]: please rename or add a comment explaining what a non-eq name is

File: src/test/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServerTest.java [hunk 4]
```diff
@@ -81,14 +79,13 @@ public void testPrepareCommand_allFlags() throws AppEngineException, ProcessRunn
...
-            "--use_mtime_file_watcher", "--threadsafe_override", "default:False,backend:True",
-            "--python_startup_script", "script.py", "--python_startup_args", "arguments",
-            "--jvm_flag", "-Dtomato", "--jvm_flag", "-Dpotato", "--custom_entrypoint", "entrypoint",
-            "--runtime", "java", "--allow_skipped_files", "--api_port", "8091",
-            "--automatic_restart", "--dev_appserver_log_level", "info",
-            "--skip_sdk_update_check", "--default_gcs_bucket_name", "buckets");
+            "--admin_port", "8000", "--auth_domain=example.com", "--storage_path=storage/path",
+            "--log_level=debug", "--max_module_instances=3", "--use_mtime_file_watcher",
+            "--threadsafe_override=default:False,backend:True", "--python_startup_script=script.py",
+            "--python_startup_args=arguments", "--jvm_flag=-Dtomato", "--jvm_flag=-Dpotato",
```
Comment (elharo) [inline src/test/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServerTest.java]: -Dtomato? -Dpotato?

Comment (meltsufin) [inline src/test/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServerTest.java]: Weren't these added Java 9? :smiley:

**Comment by meltsufin** [comment]
@elharo Tried to address your comments in the last commit.

**Comment by loosebazooka** [comment]
do we know why some are equals and some are spaces?

**Comment by meltsufin** [comment]
Not really, but it looks like the common params are space separated, while the additional ones are not.

File: src/test/java/com/google/cloud/tools/appengine/cloudsdk/internal/args/DevAppServerArgsTest.java [hunk 5]
```diff
@@ -56,7 +58,9 @@ public void testGet_boolean() {
 
   @Test
   public void testGet_integer() {
-    assertEquals(DevAppServerArgs.get("name1", 1), Arrays.asList("--name1", "1"));
-    assertEquals(DevAppServerArgs.get("name2", 2), Arrays.asList("--name2", "2"));
+    assertEquals(DevAppServerArgs.get("name1", 1), Arrays.asList("--name1=1"));
```
Comment (loosebazooka) [inline src/test/java/com/google/cloud/tools/appengine/cloudsdk/internal/args/DevAppServerArgsTest.java]: I think expected goes first?

Comment (meltsufin) [inline src/test/java/com/google/cloud/tools/appengine/cloudsdk/internal/args/DevAppServerArgsTest.java]: Yep.

**Comment by meltsufin** [comment]
It appears that you can use '=' for all dev_appserver params. @loosebazooka PTAL

**Comment by loosebazooka** [comment]
:ship:

**Pull request merged**